docs/csharp/tutorials/exploration/csharp-7.yml

```diff
@@ -0,0 +1,536 @@
+### YamlMime:Tutorial
+title: Explore C# 7 - C# interactive C# tutorial
+metadata:
+  title: Explore C# 7 - Try the new features in C# 6 interactively, using your browser
+  description: In this tutorial, you'll use your browser to explore C# 6 interactively. You'll explore the new idioms you can use with C# 6 that enable more concise and readable code.
+  audience: Developer
+  level: intermediate
+  ms.date: 03/20/2019
+  displayType: two-column
+  interactive: csharp
+items:
+- durationInMinutes: 1
+  content: |
+    This tutorial lets you explore C# 7 features interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that modify earlier C# practices to use newer, more concise C# 6 features. The rest of this article provides an overview of each of these features, with a link to explore each feature.
+
+- title: Our variable declarations at the assignment location
+  durationInMinutes: 2
+  content: |
+    The existing syntax that supports `out` parameters has been improved
+    in this version.  
+
+    Previously, you would need to separate the declaration of the out variable
+    and its initialization into two different statements:
+
+    [!code-csharp[OutVariableOldStyle](~/samples/snippets/csharp/new-in-7/program.cs#03_OutVariableOldStyle "classic out variable declaration")]
+
+    You can now declare `out` variables in the argument list of a method call,
+    rather than writing a separate declaration statement:
+
+    [!code-csharp[OutVariableDeclarations](~/samples/snippets/csharp/new-in-7/program.cs#01_OutVariableDeclarations "Out variable declarations")]
+
+    You may want to specify the type of the `out` variable for clarity,
+    as shown above. However, the language does support using an implicitly
+    typed local variable:
+
+    [!code-csharp[OutVarVariableDeclarations](~/samples/snippets/csharp/new-in-7/program.cs#02_OutVarVariableDeclarations "Implicitly typed Out variable")]
+
+    * The code is easier to read. 
+        - You declare the out variable where you use it, not on another line above.
+    * No need to assign an initial value.
+        - By declaring the `out` variable where it is used in a method call, you can't accidentally use it before it is assigned.
+
+    The most common use for this feature will be the `Try` pattern. In this
+    pattern, a method returns a `bool` indicating success or failure and an
+    `out` variable that provides the result if the method succeeds.
+
+    When using the `out` variable declaration, the declared variable "leaks" into the outer scope of the if statement. This allows you to use the variable afterwards:
+
+    ```csharp
+    if (!int.TryParse(input, out int result))
+    {    
+        return null;
+    }
+
+    return result;
+    ```
+
+- title: Tuples create light-weight data structures
```

Comment:
You might consider adding output to some of the tuples examples so that they can run interactively.